rustls/src/session.rs

```diff
@@ -433,7 +433,7 @@ pub enum MiddleboxCCS {
 }
 
 pub struct SessionCommon {
-    pub negotiated_version: Option<ProtocolVersion>,
+    pub negotiated_version: ProtocolVersion,
```

Comment:
The term "negotiated version" refers to the version that was agreed upon. At the time where `SessionCommon` is constructed, we don't know the negotiated version, so the old code was correct so that we can return the right value `None` from `get_protocol_version()`.

In the cases where we are unwrapping this, we should instead thread the known (non-Option) negotiated version through the state machine, similar to https://github.com/ctz/rustls/pull/460.

Reply:
Thank you for the reference! The new changes follow this approach much closer.